spec/models/service_spec.rb

```diff
@@ -826,17 +826,31 @@ def create_deep_tree
     let(:service_template) { FactoryGirl.create(:service_template) }
     let(:service) { FactoryGirl.create(:service, :service_template => service_template) }
 
-    describe "#custom_actions" do
-      it "get list of custom actions from linked service template" do
-        expect(service_template).to receive(:custom_actions)
-        service.custom_actions
+    context "with template" do
+      describe "#custom_actions" do
+        it "get list of custom actions from linked service template" do
+          expect(service_template).to receive(:custom_actions)
+          service.custom_actions
+        end
+      end
+
+      describe "#custom_action_buttons" do
+        it "get list of custom action buttons from linked service template" do
+          expect(service_template).to receive(:custom_action_buttons)
+          service.custom_action_buttons
+        end
       end
     end
 
-    describe "#custom_action_buttons" do
-      it "get list of custom action buttons from linked service template" do
-        expect(service_template).to receive(:custom_action_buttons)
-        service.custom_action_buttons
+    context "without valid template" do
+      @custom_button = CustomButton.create(:name => "drewasdfafs", :description => "drewtwoasfda", :applies_to_class => "Service")
+      let(:service) { FactoryGirl.create(:service, :service_template_id => -1) }
+
+      describe "#custom_action_buttons" do
+        it "get list of custom action buttons on services" do
+          expect(service).to receive(:custom_action_buttons).and_return(@custom_button)
+          service.custom_action_buttons
```

Comment:
I think you want to re-word this `expect` so that it doesn't read like you've stubbed out `service.custom_action_buttons` since that's what you're testing.

Should be something more like:
`expect(service.custom_action_buttons).to equal(@custom_button)`

You can also change the `@custom_button` to be a let for consistency, it shouldn't have to be an instance variable. In this case, since lets are lazily loaded, you can use `let!` if you need the button to be created before the spec gets ran.